db/migrate/20170927135007_migrate_dialog_field_associations_to_use_new_relationship.rb

```diff
@@ -0,0 +1,44 @@
+class MigrateDialogFieldAssociationsToUseNewRelationship < ActiveRecord::Migration[5.0]
+  class DialogFieldAssociation < ActiveRecord::Base
+    include ReservedMigrationMixin
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class DialogField < ActiveRecord::Base
+    include ReservedMigrationMixin
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class Dialog < ActiveRecord::Base
+    include ReservedMigrationMixin
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class DialogTab < ActiveRecord::Base
+    include ReservedMigrationMixin
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class DialogGroup < ActiveRecord::Base
+    include ReservedMigrationMixin
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+
+  def up
+    dialog_fields_with_associations = DialogField.all.select { |df| df.auto_refresh || df.trigger_auto_refresh }.sort { |n| -n.position }
```

Comment:
Sorting the DialogFields by the `position` column is only taking into account the field position within the group box that it is in.  Tabs and groups also have a `position` column so it is easy to rearrange the different dialog elements.  What you are really looking for here is an overall field position which would need to take into account the position of the tab and the group that the field is in.

Does it make sense to add a calculation method in these stub methods to calculate that distinct position?
Something like:  `(tab.position * 1000) + (group.position * 100) + field.position`